File: java/src/main/java/mx/iteso/Cone.java
package mx.iteso;

import mx.iteso.icecreams.IceCream;
/**
 * Clase cono, parte del helado.
 */
public class Cone {
    /**Helado. */
    private IceCream iceCream;
    /**
     * Metodo para preparar el cono.
     */
    public void prepare() {
        System.out.println("Preparando cono...");
    }
    /**
     * @param newIceCream valor del helado.
     */
    public void addIceCream(final IceCream newIceCream) {
        this.iceCream = newIceCream;
    }

    /**Sirviendo el cono con helado. */
    public void serve() {
        System.out.println("Serving...");
    }
    /**
     * @return helado actual.
     */
    public IceCream getIceCream() {
        return this.iceCream;
    }
}

File: csharp/factory-method/Cone.cs
using System;
using factorymethod.main.icecreams;

namespace factorymethod.main
{
    internal class Cone
    {
        private IceCream ic;

        public static string prepare()
        {
            Console.WriteLine("Preparing Cone");
        }

        public void addIceCream(IceCream iceCream)
        {
            this.ic = iceCream;
        }

        public string serve()
        {
            Console.WriteLine("Sirviendo" );
        }

        public IceCream getIceCream()
        {
            return this.ic;
        }
    }

}